account_move_line_base_tax/account.py

```diff
@@ -49,30 +51,33 @@ def onchange_tax_secondary(self, cr, uid, ids, account_id=False, context=None):
         else:
             return {'value': {}}
 
-    def write(self, cr, uid, ids, vals, context=None, check=True, update_check=True):
+    def write(self, cr, uid, ids, vals, context=None, check=True,
+              update_check=True):
         if context is None:
             context = {}
         if not ids:
             return True
         if isinstance(ids, (int, long)):
             ids = [ids]
 
-        res = super(AccountMoveLine, self).write(cr, uid, ids, vals,
-                                                 context=context, check=check, update_check=update_check)
+        res = super(AccountMoveLine, self).write(
+            cr, uid, ids, vals, context=context, check=check,
+            update_check=update_check)
         for line in self.browse(cr, uid, ids, context=context):
-            if line.tax_id_secondary and line.tax_id_secondary.type_tax_use == 'purchase':
-                cat_tax = line.tax_id_secondary.tax_category_id
-                if cat_tax and cat_tax.name in ('IVA', 'IVA-EXENTO') and line.amount_base <= 0 and\
-                        not line.not_move_diot:
-                    raise osv.except_osv(_('Warning!'), _('The lines with tax of purchase, need '
-                                                          'have a value in the amount base.'))
-                elif cat_tax and cat_tax.name == 'IVA-RET' and line.credit <= 0 and\
-                        not line.not_move_diot:
-                    raise osv.except_osv(_('Warning!'), _('The lines with tax of purchase, need '
-                                                          'have a value in the credit.'))
+            if (not line.tax_id_secondary or
+                    line.tax_id_secondary.type_tax_use != 'purchase'):
+                continue
+            cat_tax = line.tax_id_secondary.tax_category_id
+            if all([cat_tax, cat_tax.name == 'IVA-RET', line.credit <= 0,
```

Comment:
I left you a comment in another PR but with the a similar request that I want to see here:

https://github.com/Vauxoo/odoo-mexico-v2/pull/390#discussion_r113048608